teacher.rb

```diff
@@ -0,0 +1,12 @@
+class Teacher < Person
+  # constructor
+  def initialized(specialization)
+    super(id, age, name, parent_permission: true)
+    @specialization = specialization
+  end
```

Comment:
- [ ] Here also smile, the constructor should take four parameters - age, specialization, name, and parent_permission. The age and classroom parameters are mandatory, while name has a default value of "Unknown," and parent_permission has a default value of true. This allows for flexibility when creating instances of the class as I said previously 👍 . Kindly update the Teacher Constructor 👍

```ruby
  def initialize(age, specialization, name = "Unknown", parent_permission = true)
    super(age, name, parent_permission)
    @specialization = specialization
  end
```